src/main/java/edu/tamu/cap/service/FedoraService.java

```diff
@@ -263,9 +281,18 @@ public RepositoryViewContext createMetadata(String contextUri,Triple triple) thr
         FcrepoResponse response = patch.perform();
         URI location = response.getLocation();
         logger.debug("Metadata creation status and location: {}, {}", response.getStatusCode(), location);
+        messagingService.sendMessage(MESSAGING_CHANNEL, extractContextPath(contextUri, METADATA_CREATE));
```

Comment:
I am concerned about these messages being generated from within the fedora service. The fact that they are occurring becomes a feature of this specific implementation and not a general feature of all RepositoryViewService. Would it be possible to handle this in a similar way to the other feature sets (i.e. implementing a messaging interface, and possibly using an aspect to target it?)

If we did this then all RepositoryViewService implementations could benefit from this instead of just the fedora impl.

Reply:
https://github.com/TAMULib/Cap/commit/ea16b679a27b5ad9cea9fd147e6a45571193728f